lib/ocpp/v201/device_model.cpp

```diff
@@ -367,5 +367,98 @@ void DeviceModel::check_integrity(const std::map<int32_t, int32_t>& evse_connect
     }
 }
 
+std::vector<SetMonitoringResult> DeviceModel::set_monitors(const std::vector<SetMonitoringData>& requests) {
+    if (requests.size() <= 0) {
+        return {};
+    }
+
+    std::vector<SetMonitoringResult> set_monitors_res;
+
+    for (auto& request : requests) {
+        SetMonitoringResult result;
+
+        if (this->device_model.find(request.component) == this->device_model.end()) {
+            result.status = SetMonitoringStatusEnum::UnknownComponent;
+            set_monitors_res.push_back(result);
+            continue;
+        }
+
+        auto variable_map = this->device_model[request.component];
+
+        if (variable_map.find(request.variable) == variable_map.end()) {
+            result.status = SetMonitoringStatusEnum::UnknownVariable;
+            set_monitors_res.push_back(result);
+            continue;
+        }
+
+        // TODO (ioan): see how we should handle the 'Duplicate' data
+        if (this->storage->set_monitoring_data(request)) {
```

Comment:
After inserting the monitor in the storage to store it persistently, we should set it memory in the monitors map